thoughts-of-a-rustacean-learning-go.md

```diff
@@ -0,0 +1,244 @@
+<!--
+headline: Thoughts of a Rustacean learning Go
+description: the experiences of a Rustacean learning Go
+author: Manish Goregaokar
+datePublished: 2015-04-24
+twitter: ManishEarth
+github: manishearth
+-->
+
+// Might want to mention that this is a crosspost
+
+
+For the past year or so I've been playing with the [Rust programming language][trpl].
+I've come to love the language and spend quite a bit of my free time contributing to the [compiler][compiler]
+and the [Servo browser engine][servo].
+
+// Perhaps a different form of introduction?
+
+
+[trpl]: http://rust-lang.org/
+[compiler]: https://github.com/rust-lang/rust
+[servo]: https://github.com/servo/servo
+
+Recently, for a course, we were told to use Go for the assignments and project. Whilst a tad disappointed
+that I couldn't use Rust for it, I saw this as a great opportunity. Rust and Go have been
+compared a lot as the "hot new languages", and finally I'd get to see the other side of the argument.
+
+By the end of the course, aside from some simpler assignments, I'd written
+[a basic implementation of the Raft consensus algorithm][raft] and a [toy application][dash] using it,
+both in Go.
+
+Before I get into the experience, let me preface this by mentioning that Rust and Go don't exactly target the same audiences.
+Go is garbage collected and is okay with losing out on some performance for ergonomics;
+whereas Rust tries to keep everything as a compile time check as much as possible.
+This usually makes Rust much more useful for lower level applications.
```

Comment:
I might make this statement a bit less definitive, e.g., "This can make Rust better for lower level applications".

Is there something we could reference in this paragraph, perhaps:

* some performance benchmarks.
* or another comparative article, talking about the strengths of both languages?